examples/metrics/kafka-connect-metrics.yaml

```diff
@@ -8,113 +8,40 @@ spec:
-        name: kafka_$2_$5
-        labels:
-          clientId: "$3"
-          nodeId: "$4"
-        help: "Kafka $1 JMX metric type $2"
-        type: UNTYPED
-
-      #kafka.connect:type=kafka-metrics-count,client-id="{clientid}"
-      #kafka.consumer:type=consumer-fetch-manager-metrics,client-id="{clientid}"
-      #kafka.consumer:type=consumer-coordinator-metrics,client-id="{clientid}"
-      #kafka.consumer:type=consumer-metrics,client-id="{clientid}"
-      - pattern: kafka.(.+)<type=(.+)-metrics, client-id=(.*)><>(.+-total|.+-avg|.+-bytes|.+-count|.+-ratio|.+-age|.+-flight|.+-threads|.+-connectors|.+-tasks|.+-ago)
-        name: kafka_$2_$4
-        labels:
-          clientId: "$3"
-        help: "Kafka $1 JMX metric type $2"
-        type: GAUGE
-
-      #kafka.connect:type=connector-task-metrics,connector="{connector}",task="{task}<> status"
-      - pattern: 'kafka.connect<type=connector-task-metrics, connector=(.+), task=(.+)><>status: ([a-z-]+)'
-        name: kafka_connect_connector_status
-        value: 1
-        labels:
-          connector: "$1"
-          task: "$2"
-          status: "$3"
-        help: "Kafka Connect JMX Connector status"
-        type: GAUGE
-
-      #kafka.connect:type=task-error-metrics,connector="{connector}",task="{task}"
-      #kafka.connect:type=source-task-metrics,connector="{connector}",task="{task}"
-      #kafka.connect:type=sink-task-metrics,connector="{connector}",task="{task}"
-      #kafka.connect:type=connector-task-metrics,connector="{connector}",task="{task}"
-      - pattern: kafka.connect<type=(.+)-metrics, connector=(.+), task=(.+)><>(.+-total|.+-count|.+-ms|.+-ratio|.+-avg|.+-failures|.+-requests|.+-timestamp|.+-logged|.+-errors|.+-retries|.+-skipped)
-        name: kafka_connect_$1_$4
-        labels:
-          connector: "$2"
-          task: "$3"
-        help: "Kafka Connect JMX metric type $1"
-        type: GAUGE
-
-      #kafka.connect:type=connector-metrics,connector="{connector}"
-      #kafka.connect:type=connect-worker-metrics,connector="{connector}"
-      - pattern: kafka.connect<type=connect-worker-metrics, connector=(.+)><>([a-z-]+)
-        name: kafka_connect_worker_$2
-        labels:
-          connector: "$1"
-        help: "Kafka Connect JMX metric $1"
-        type: GAUGE
-
-      #kafka.connect:type=connect-worker-metrics
-      - pattern: kafka.connect<type=connect-worker-metrics><>([a-z-]+)
-        name: kafka_connect_worker_$1
-        help: "Kafka Connect JMX metric worker"
-        type: GAUGE
+  metricsConfig:
+    type: jmxPrometheusExporter
+    valueFrom:
+      configMapKeyRef:
+        name: connect-metrics
+        key: metrics-config.yml
+
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: connect-metrics
+  labels:
+    app: strimzi
+data:
+  metrics-config.yml: '{"lowercaseOutputLabelNames":true,"lowercaseOutputName":true,"rules":[{"help":"Kafka
```

Comment:
Is it YAML or JSON? Looks like JSON to me, so the file should ba probably named accordingly?

Also, does this mean users will have to convert the metrics configuration they have instead of just copy-pasting it? That would be rather inconvenient 🤔

In any case you should make the example file nicer and prettyprint the json to make it easily readable and maybe use something like multiline string.